poseidon2.nim

```diff
@@ -56,13 +16,64 @@ func permInplace*(x, y, z : var F) =
   for j in 4..7:
     externalRound(j, x, y, z)
 
+# the Poseidon2 permutation
 func perm*(xyz: S) : S =
   var (x,y,z) = xyz
   permInplace(x, y, z)
   return (x,y,z)
 
 #-------------------------------------------------------------------------------
 
+# sponge with rate=1 (capacity=2) 
+func spongeWithRate1*(xs: openArray[F]) : F =
```

Comment:
Will `xs` be fixed or arbitrary size?

Reply:
arbitrary, this is a "traditional" hash function.